### DISCUSSION TITLE
Models: `autoincrement()` optionnally starting at 0 ?

### DISCUSSION THREAD

**Discussion by cassepipe** [prisma/prisma#18246] [Q&A] [answered]
Is it possible to have `@id` + `autoincrement()` start at 0 instead of 1 ?

**Comment by ludralph** [comment, marked as answer]
Hi @cassepipe 👋 

It's not possible from the schema directly but you can modify the migration file to achieve this. This example uses a Postgres database.

Let's assume that we have to start the Task id from 0. To do that, Add the model in `schema.prisma` file
```js
model Task {
  id        Int   @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String?
}
```
Then run the below command
```
 npx prisma  migrate dev --create-only
```
It will add the migration file in the prisma/migrations folder. Then open the migration.sql file and add the following code
```sql
ALTER SEQUENCE "Task_id_seq" MINVALUE 0 START 0 RESTART 0;
```
Then run the following command to apply in the database
```
 npx prisma  migrate dev
```
Then you should see the output like below
<img width="651" alt="image" src="https://user-images.githubusercontent.com/13855608/223902965-bd34805d-96bc-4060-b6d7-c0ec595b6c60.png">
From here, you can then proceed to create records. For example
```js
const res = await prisma.task.create({
  data: {
    title: "testing",
    updatedAt: new Date()
  }
 })
```
Logging the result to the console, you should see the id starting at 0
```js
{
  id: 0,
  createdAt: 2023-03-09T02:35:53.587Z,
  updatedAt: 2023-03-09T02:35:53.532Z,
  title: 'testing'
}
```
I hope this helps. Feel free to let me know if you have any other questions.

**Reply by cassepipe** [reply]
Hi @ludralph 
Thank you very much for the helpful answer. 
Under the guise of generality, I was actually even more interested in `postgresql` more specifically, so thanks again.
I have taken the liberty to ask the question on Stack Overflow as weel as I thought an eventual answer would be more discoverable to others.
Here is the link to the SO question : https://stackoverflow.com/questions/75663635/prisma-models-can-autoincrement-start-at-0/75679539#75679539
Would you like to copy your answer over there ? Or would you agree that I do it ?

**Reply by cassepipe** [reply]
For the purpose of knowledge sharing, I have taken the liberty to post your answer under my SO question  (crediting the answer to you)
Let me know if you want it taken down or if you want to post the answer yourself with your SO account and get the reputation it deserves.
